### PR TITLE
net: openthread: make diag compile

### PR DESCRIPTION
**Pull request by pszkotak**
Fix compilation error when the DIAG module is enabled.
All core functionalities are provided.

Signed-off-by: Piotr Szkotak <piotr.szkotak@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/openthread/platform/diag.c [hunk 1]
```diff
@@ -17,18 +15,22 @@
  */
 static bool sDiagMode;
 
-void otPlatDiagProcess(otInstance *aInstance,
-		       int argc,
-		       char *argv[],
-		       char *aOutput,
-		       size_t aOutputMaxLen)
+otError otPlatDiagProcess(otInstance *aInstance,
+					uint8_t argc,
```
Comment (rlubos) [inline subsys/net/lib/openthread/platform/diag.c:19]: Please fix the alignment. Note, it's 8 spaces per tab, I guess hence the confusion.

File: subsys/net/lib/openthread/platform/diag.c [hunk 2]
```diff
@@ -52,8 +54,8 @@ void otPlatDiagTxPowerSet(int8_t aTxPower)
 }
 
 void otPlatDiagRadioReceived(otInstance *aInstance,
-			     RadioPacket *aFrame,
-			     ThreadError aError)
+						otRadioFrame *aFrame,
```
Comment (rlubos) [inline subsys/net/lib/openthread/platform/diag.c:57]: Please fix the alignment.

File: subsys/net/lib/openthread/platform/diag.c [hunk 3]
```diff
@@ -17,18 +15,22 @@
...
 	ARG_UNUSED(argc);
 	ARG_UNUSED(aInstance);
 
+	otError error = OT_ERROR_NOT_IMPLEMENTED;
+
 	/* Add more plarform specific diagnostics features here. */
 	snprintk(aOutput, aOutputMaxLen,
 		 "diag feature '%s' is not supported\r\n", argv[0]);
+
+	return error;
```
Comment (rlubos) [inline subsys/net/lib/openthread/platform/diag.c:33]: I'd return `OT_ERROR_NOT_IMPLEMENTED` directly here, no need for a variable.

**Review by rlubos — CHANGES_REQUESTED** [review]
Please prefix the commit title with `net: openthread:` instead. Otherwise just some nits.

**Review by rlubos — APPROVED** [review]
LGTM, thanks!

**Review by jukkar — APPROVED** [review]

**Pull request merged**